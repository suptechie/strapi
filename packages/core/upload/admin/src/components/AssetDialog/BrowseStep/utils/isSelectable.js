/**
 * @typedef isSelectable
 *
 * @type {allowedTypes: Array<string>}
 * @type {mime: string}
 * @returns {Boolean}
 */

export const isSelectable = (allowedTypes, mime = '') => {
  if (!mime) return false;

  const fileType = mime.split('/')[0];

  return (
    allowedTypes.includes(fileType) ||
    (allowedTypes.includes('file') && !['video', 'image', 'audio'].includes(fileType))
  );
};
