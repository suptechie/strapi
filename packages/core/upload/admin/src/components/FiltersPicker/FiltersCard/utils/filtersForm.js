import { dateToUtcTime } from '@strapi/helper-plugin';
import moment from 'moment';

const filtersForm = {
  createdAt: {
    type: 'datetime',
    defaultFilter: '=',
    defaultValue: dateToUtcTime(moment()),
  },
  updatedAt: {
    type: 'datetime',
    defaultFilter: '=',
    defaultValue: dateToUtcTime(moment()),
  },
  size: {
    type: 'integer',
    defaultFilter: '=',
    defaultValue: '0KB',
  },
  mime: {
    type: 'enum',
    defaultFilter: '_contains',
    defaultValue: 'image',
  },
};

export default filtersForm;
