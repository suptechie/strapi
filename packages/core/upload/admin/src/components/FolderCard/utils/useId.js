import { useRef } from 'react';

let id = 0;

const genId = () => ++id;

const useId = (prefix, initialId) => {
  const idRef = useRef(initialId || `${prefix}-${genId()}`);

  return idRef.current;
};

export default useId;
