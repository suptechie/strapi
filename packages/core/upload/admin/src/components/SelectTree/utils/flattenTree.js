export default function flattenTree(tree, parent, depth = 1) {
  return tree.flatMap(item =>
    item.children
      ? [{ ...item, parent: parent?.value, depth }, ...flattenTree(item.children, item, depth + 1)]
      : { ...item, depth, parent: parent?.value }
  );
}
