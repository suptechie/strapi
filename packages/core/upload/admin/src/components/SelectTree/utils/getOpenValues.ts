interface Option {
  value: number | string | null;
  parent?: number | string | null;
}

interface DefaultValue {
  value?: number | string | null;
}

function getOpenValues(options: Option[], defaultValue: DefaultValue = {}) {
  let values: Option['value'][] = [];
  const { value } = defaultValue;
  const option = options.find((option) => option.value === value);

  if (!option) {
    return values;
  }

  values.push(option.value);

  let { parent } = option;

  while (parent !== undefined) {
    const option = options.find(({ value }) => value === parent);

    if (!option) {
      break;
    }

    values.push(option.value);
    parent = option.parent;
  }

  return values.reverse();
}

export default getOpenValues;
