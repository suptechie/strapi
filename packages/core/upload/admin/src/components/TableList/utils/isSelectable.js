export const isSelectable = (allowedTypes, elementType, fileType, isFolderSelectionAllowed) => {
  let canSelectElement;

  if (elementType === 'folder') {
    canSelectElement = isFolderSelectionAllowed;
  } else {
    canSelectElement =
      allowedTypes.includes(fileType) ||
      (allowedTypes.includes('file') && !['video', 'image', 'audio'].includes(fileType));
  }

  return canSelectElement;
};
