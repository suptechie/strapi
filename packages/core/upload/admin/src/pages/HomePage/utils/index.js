export { default as getHeaderLabel } from './getHeaderLabel';
