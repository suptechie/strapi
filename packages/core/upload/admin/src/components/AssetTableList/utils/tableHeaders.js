import { getTrad } from '../../../utils';

export const tableHeaders = [
  {
    name: 'preview',
    key: 'preview',
    metadatas: {
      label: { id: getTrad('table-header-preview'), defaultMessage: 'preview' },
      sortable: false,
    },
  },
  {
    name: 'name',
    key: 'name',
    metadatas: {
      label: { id: getTrad('table-header-name'), defaultMessage: 'name' },
      sortable: true,
    },
  },
  {
    name: 'extension',
    key: 'extension',
    metadatas: {
      label: { id: getTrad('table-header-extension'), defaultMessage: 'extension' },
      sortable: false,
    },
  },
  {
    name: 'size',
    key: 'size',
    metadatas: {
      label: { id: getTrad('table-header-size'), defaultMessage: 'size' },
      sortable: false,
    },
  },
  {
    name: 'createdAt',
    key: 'createdAt',
    metadatas: {
      label: { id: getTrad('table-header-createdAt'), defaultMessage: 'Created' },
      sortable: true,
    },
  },
  {
    name: 'updatedAt',
    key: 'updatedAt',
    metadatas: {
      label: { id: getTrad('table-header-updatedAt'), defaultMessage: 'Last update' },
      sortable: true,
    },
  },
];
