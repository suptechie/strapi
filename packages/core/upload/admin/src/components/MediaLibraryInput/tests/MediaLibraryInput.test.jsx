import React from 'react';

import { Form } from '@strapi/admin/strapi-admin';
import { screen, render } from '@tests/utils';

import { MediaLibraryInput } from '../index';

describe('<MediaLibraryInput />', () => {
  it('renders and matches the snapshot', () => {
    render(<MediaLibraryInput name="test" label="default message" required />, {
      renderOptions: {
        wrapper: ({ children }) => (
          <Form onSubmit={jest.fn()} method="POST">
            {children}
          </Form>
        ),
      },
    });

    expect(screen.getByRole('button')).toMatchInlineSnapshot(`
      .c0 {
        width: 100%;
        height: 100%;
      }

      .c2 {
        font-size: 1.2rem;
        line-height: 1.33;
        font-weight: 600;
        color: #666687;
      }

      .c1 {
        align-items: center;
        display: flex;
        flex-direction: column;
        gap: 12px;
        justify-content: center;
      }

      .c3 {
        align-items: center;
      }

      <button
        class="c0 c1"
        style="cursor: pointer;"
        type="button"
      >
        <svg
          aria-hidden="true"
          fill="#4945ff"
          height="3.2rem"
          viewBox="0 0 32 32"
          width="3.2rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 3a13 13 0 1 0 13 13A13.016 13.016 0 0 0 16 3m5 14h-4v4a1 1 0 0 1-2 0v-4h-4a1 1 0 0 1 0-2h4v-4a1 1 0 0 1 2 0v4h4a1 1 0 0 1 0 2"
          />
        </svg>
        <span
          class="c2 c3"
          style="text-align: center;"
        >
          Click to add an asset or drag and drop one in this area
        </span>
      </button>
    `);
  });
});
