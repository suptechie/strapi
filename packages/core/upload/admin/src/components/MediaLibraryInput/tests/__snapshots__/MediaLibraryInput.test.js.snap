// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MediaLibraryInput /> renders and matches the snapshot 1`] = `
.c16 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c10 {
  width: 100%;
  height: 100%;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c12 {
  color: #4945ff;
  margin-bottom: 12px;
  width: 30px;
  height: 24px;
}

.c13 path {
  fill: #4945ff;
}

.c14 {
  font-weight: 600;
  color: #666687;
  font-size: 0.75rem;
  line-height: 1.33;
}

.c0 {
  padding-bottom: 4px;
}

.c2 {
  background: #f6f6f9;
  padding: 8px;
  border-radius: 4px;
  border-color: #dcdce4;
  border: 1px solid #dcdce4;
}

.c3 {
  position: relative;
}

.c5 {
  padding-right: 8px;
  padding-left: 8px;
  width: 100%;
}

.c7 {
  height: 124px;
}

.c1 {
  font-weight: 600;
  color: #32324d;
  font-size: 0.75rem;
  line-height: 1.33;
}

.c4 {
  display: grid;
  grid-template-columns: auto 1fr auto;
  grid-template-areas: 'startAction slides endAction';
}

.c6 {
  grid-area: slides;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c15 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<body>
  <div
    class=""
  >
    <div
      class="c0"
    >
      <span
        class="c1"
      >
        default
      </span>
    </div>
    <div
      class="c2"
    >
      <section
        aria-label="default"
        aria-roledescription="carousel"
        class="c3 c4"
      >
        <div
          aria-live="polite"
          class="c5 c6"
          width="100%"
        >
          <div
            aria-label="upload.mediaLibraryInput.slideCount"
            aria-roledescription="slide"
            class="c7 c8 c9"
            height="124px"
            role="group"
          >
            <button
              class="c10 c11"
              height="100%"
              style="cursor: pointer;"
              type="button"
              width="100%"
            >
              <svg
                aria-hidden="true"
                class="c12 c13"
                fill="none"
                height="24px"
                viewBox="0 0 24 20"
                width="30px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21.525 2.394H7.814v1.582h13.711c.47 0 .824.498.824 1.091v9.834l-2.785-3.165c-.2-.233-.453-.42-.738-.549a2.21 2.21 0 00-.91-.194 2.209 2.209 0 00-.91.19 2.126 2.126 0 00-.739.545l-2.925 3.378-2.967-3.354a2.143 2.143 0 00-.739-.543 2.226 2.226 0 00-.909-.193c-.315.002-.626.07-.911.2-.285.13-.536.318-.737.552l-2.925 3.425V7.637H2.505v9.874c.023.153.059.304.107.451v.127l.042.095c.056.142.125.28.206.411l.099.15c.074.106.157.207.247.301l.123.119c.13.118.274.221.429.308h.025c.358.214.772.327 1.195.325h16.481a2.47 2.47 0 001.037-.239c.322-.153.605-.375.826-.648.222-.263.392-.564.502-.886.067-.216.108-.439.124-.664V5.067a2.493 2.493 0 00-.65-1.805 2.7 2.7 0 00-1.773-.868z"
                  fill="#32324D"
                />
                <path
                  d="M12.526 9.18c.91 0 1.648-.708 1.648-1.582 0-.874-.738-1.582-1.648-1.582-.91 0-1.648.708-1.648 1.582 0 .874.738 1.582 1.648 1.582zM3.297 6.395h.823V3.988h2.473v-.79-.792H4.12V0h-.823c-.219 0-.849.002-.825 0v2.406H0v1.582h2.472v2.407h.825z"
                  fill="#32324D"
                />
              </svg>
              <span
                class="c14 c15"
                style="text-align: center;"
              >
                upload.mediaLibraryInput.placeholder
              </span>
            </button>
          </div>
        </div>
      </section>
    </div>
  </div>
  <div
    class="c16"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</body>
`;
